### ISSUE TITLE
highlight: local WinSeparator should highlight all borders (unless adjacent window has local WinSeparator)

### ISSUE THREAD

**Issue by musjj** [neovim/neovim#24266]
### Problem

When setting the local `winhighlight` for `WinSeparator`, only the right border of the window will be affected.

### Steps to reproduce

```sh
nvim --clean
:e /tmp/foo
<C-w>v
:e /tmp/bar
:setlocal winhighlight=WinSeparator:Error
<C-w>L
```

### Expected behavior

The window separator for `/tmp/bar` should stay red, even after you move the window to the right with `<C-w>L`.

### Neovim version (nvim -v)

v0.10.0-dev-606+g628f6cce8

### Vim (not Nvim) behaves the same?

vim does not have the `winhighlight` option

### Operating system/version

Arch Linux

### Terminal name/version

kitty 0.28.1

### $TERM environment variable

xterm-kitty

### Installation

bob

**Comment by zeertzjq** [comment]
> The window separator for `/tmp/bar` should stay red, even after you move the window to the right with `<C-w>L`.

Why?

**Comment by musjj** [comment]
Hmmm, my intuition would say that any separator touching the window should be affected by its local highlight. Though this does raise the issue of conflict handling (first come first served?).

In fact, is there a reason why only the *right* border is affected (instead of any other one)?

**Comment by justinmk** [comment]
> is there a reason why only the _right_ border is affected (instead of any other one)?

Because split separators are a single column, they can't be divided between adjacent windows. This is just a vim (TUI) limitation.

**Comment by bfredl** [comment]
We had to pick one or the other due to the conflict, and this was the one that fit best with how the code works internally.

**Comment by musjj** [comment]
> Because split separators are a single column, they can't be divided between adjacent windows. This is just a vim (TUI) limitation.

Yeah, I've realized this, which is why I mentioned the potential conflict handling.

Can this issue be re-opened as a feature request, or is this considered a `wontfix`?

My use-case is that this allows "docked" windows that has a different `Normal` color to blend better with its border. The current behavior works well for file explorer plugins that docks to the left by default, but not so much for plugins that docks to the bottom or any other direction.

**Comment by justinmk** [comment]
> Expected behavior
> The window separator for `/tmp/bar` should stay red, even after you move the window to the right with `<C-w>L`.

But that means the other window's border is overridden.

> Can this issue be re-opened as a feature request, or is this considered a `wontfix`?

What is the request? Perhaps this:

- If 'winhighlight' is set on a window, its `WinSeparator` highlights _all_ borders.
- If 'winhighlight' is set on multiple adjacent windows how is the priority decided?

**Comment by musjj** [comment]
> If 'winhighlight' is set on a window, its WinSeparator highlights all borders.

Yes.

> If 'winhighlight' is set on multiple adjacent windows how is the priority decided?

Yeah, I think this should be discussed. Not sure what's the simplest way of resolving the conflict. I was thinking that it should be based on when `setlocal` was called (the newest `setlocal` takes priority). But does that mean that you need to store additional timestamp metadata? Or is there a simpler heuristic?

What do you think?
